### PR TITLE
Implement resume exporter module with JSON export and validation tests 

### LINKED ISSUE #111: Generate resume items

### PR DESCRIPTION
**Pull request by Puneet-Maan**
## Description

Implemented the resume exporter module which builds on the existing resume item generator to produce a consolidated JSON export of all detected projects. This component discovers visible project directories, constructs ResumeItem dataclasses for each, and writes a timestamped, deterministic JSON payload suitable for resume generation or downstream processing.

Key additions:
- resume_exporter.py exports all project resume data to JSON with a CLI interface  
- test_resume_exporter.py has functional tests for project discovery and export behavior  
- test_resume_exporter_json_validation.py has validation tests for JSON structure, determinism, and CLI output  
- Project ordering and stable export structure  
- Timezone aware ISO8601 timestamps in JSON output  
- Optional schema validation if jsonschema is available

Closes #111 

---

## Type of Change

- [ ] Bug fix  
- [x] New feature  
- [ ] Breaking change  
- [ ] Documentation update  
- [x] Tests added or updated  
- [ ] Refactoring  
- [ ] Performance improvement  

---

## Testing

The following automated test suites validate functionality:

| Test File | Purpose |
|---|---|
| test_resume_exporter.py | Tests project discovery, dataclass creation, and JSON export content |
| test_resume_exporter_json_validation.py | Tests JSON structure, deterministic output, CLI behavior, and optional schema validation |

Key test coverage includes:
- Hidden directory and file exclusion (.git, non-project files)  
- Verification of exported JSON structure and timestamps  
- Framework and highlight detection from requirements.txt 
- Deterministic results across repeated runs  
- End to end CLI export invocation 

Executed using:
python3 -m unittest discover -s test -p "test_resume_exporter.py" -v
python3 -m unittest discover -s test -p "test_resume_exporter_json_validation.py" -v

Manual verification:
- Verified inclusion of framework specific highlights
- Confirmed valid JSON payload written to file

---

##  Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [x] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [x] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

image

image

</details>

### CODE REVIEW

File: src/resume_exporter.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        "-o",
+        "--output",
+        dest="output",
+        help="Destination JSON file (defaults to <root>/resume_items.json).",
+    )
+
+    args = parser.parse_args(argv)
+    export_resume_items(args.root, args.output)
+    return 0
+
```
[the remark this answers is not in the document]
Comment (Puneet-Maan) [reply src/resume_exporter.py]: All good, this is a backend utility CLI for now

File: test/test_resume_exporter_json_validation.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    def test_cli_behavior(self) -> None:
+        """Ensure CLI invocation writes valid JSON file."""
+        print("[Resume JSON Test] Verifying CLI export functionality...", flush=True)
+
+        cmd = [sys.executable, "-m", "src.resume_exporter", str(self.root), "-o", str(self.output_path)]
+        proc = run(cmd, stdout=PIPE, stderr=PIPE, text=True)
+
+        self.assertEqual(proc.returncode, 0, f"CLI failed: {proc.stderr}")
+        payload = json.loads(self.output_path.read_text())
+        self.assertIn("projects", payload)
```
[the remark this answers is not in the document]
Comment (Puneet-Maan) [reply test/test_resume_exporter_json_validation.py]: I'll look at this pytest issue in the future, looks like that same windows or permission, etc. issue we have seen before in other PRs.

for the mock test, I agree! mocking user interaction would make sense once the CLI becomes more interactive.
For now, since this CLI just exports JSON without prompts, I kept it as a straightforward integration test

**Review by mahigangal — APPROVED** [review]
Great work on the PR! All tests pass and have good coverage for edge cases. Thanks for the detailed docstrings and for separating the functionality into shorter functions. Thorough work on the timestamp validation. 

I have added a couple of comments on error handling, but the PR is approved since nothing is breaking and everything looks good!

File: src/resume_exporter.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    Args:
+        root: Directory that contains one or more project folders.
+        destination: Where the JSON should be written. Defaults to
+            ``root / "resume_items.json"``.
+
+    Returns:
+        Path to the written JSON file.
+    """
+    root_path = Path(root).resolve()
```
[the remark this answers is not in the document]
Comment (Puneet-Maan) [reply src/resume_exporter.py]: good one! I’ve added a validation check to ensure the provided root_path exists and is a directory before proceeding

[the remark this answers is not in the document]
Comment (mahigangal) [reply src/resume_exporter.py]: thanks for the quick fixes! Ready to be merged.
